File: admin/flutter/lib/features/p2p_merchant/p2p_merchant_screen.dart
/**
 * TigerWallet Admin - P2P Merchant Feature
 * Complete implementation with dark/light theme support
 */

import 'package:flutter/material.dart';
import 'package:shared_preferences/shared_preferences.dart';

import '../../../core/network/api_client.dart';
import '../../../core/network/dio_client.dart';

class P2PMerchantScreen extends StatefulWidget {
  const P2PMerchantScreen({Key? key}) : super(key: key);

  @override
  State<P2PMerchantScreen> createState() => _P2PMerchantScreenState();
}

class _P2PMerchantScreenState extends State<P2PMerchantScreen> {
  List<Map<String, dynamic>> _merchants = [];
  bool _loading = true;
  String? _error;
  String _filter = 'all';
  bool _isDark = false;

  ApiClient? _api;

  @override
  void initState() {
    super.initState();
    _loadMerchants();
  }

  Future<ApiClient> _client() async {
    if (_api != null) return _api!;
    final prefs = await SharedPreferences.getInstance();
    _api = DioClient.withPrefs(prefs);
    return _api!;
  }

  Future<void> _loadMerchants() async {
    setState(() {
      _loading = true;
      _error = null;
    });
    try {
      final api = await _client();
      _merchants = await api.getAdminP2PMerchants(status: _filter);
      setState(() {});
    } catch (e) {
      setState(() {
        _error = e.toString();
        _merchants = [];
      });
    } finally {
      setState(() => _loading = false);
    }
  }

  Future<void> _approveMerchant(String id) async {
    try {
      final api = await _client();
      await api.approveP2PMerchant(id);
      _loadMerchants();
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text('Approve failed: $e')));
      }
    }
  }

  Future<void> _rejectMerchant(String id) async {
    final reason = await showDialog<String>(
      context: context,
      builder: (context) {
        final controller = TextEditingController();
        return AlertDialog(
          title: const Text('Reject Merchant'),
          content: TextField(controller: controller, decoration: const InputDecoration(labelText: 'Reason')),
          actions: [
            TextButton(onPressed: () => Navigator.pop(context), child: const Text('Cancel')),
            TextButton(onPressed: () => Navigator.pop(context, controller.text), child: const Text('Reject')),
          ],
        );
      },
    );
    if (reason != null && reason.isNotEmpty) {
      try {
        final api = await _client();
        await api.rejectP2PMerchant(id, reason);
        _loadMerchants();
      } catch (e) {
        if (mounted) {
          ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text('Reject failed: $e')));
        }
      }
    }
  }

  void _toggleTheme() => setState(() => _isDark = !_isDark);

  @override
  Widget build(BuildContext context) {
    return Theme(
      data: _isDark ? ThemeData.dark() : ThemeData.light(),
      child: Scaffold(
        appBar: AppBar(
          title: const Text('P2P Merchants'),
          actions: [IconButton(icon: Icon(_isDark ? Icons.light_mode : Icons.dark_mode), onPressed: _toggleTheme)],
        ),
        body: Column(
          children: [
            _buildStats(),
            _buildFilterChips(),
            Expanded(child: _buildMerchantsList()),
          ],
        ),
      ),
    );
  }

  Widget _buildStats() {
    return Container(
      padding: const EdgeInsets.all(16),
      child: Row(
        children: [
          Expanded(child: _buildStatCard('Total', '${_merchants.length}')),
          Expanded(child: _buildStatCard('Approved', '${_merchants.where((m) => m['status'] == 'approved').length}')),
          Expanded(child: _buildStatCard('Pending', '${_merchants.where((m) => m['status'] == 'pending').length}')),
          Expanded(
            child: _buildStatCard(
              'Volume',
              '\$${(_merchants.fold(0.0, (sum, m) => sum + ((m['total_volume'] as num?) ?? 0.0)) / 1000).toStringAsFixed(0)}K',
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildStatCard(String label, String value) {
    return Card(
      child: Padding(
        padding: const EdgeInsets.all(12),
        child: Column(
          children: [
            Text(value, style: const TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
            Text(label, style: TextStyle(fontSize: 11, color: Colors.grey[600])),
          ],
        ),
      ),
    );
  }

  Widget _buildFilterChips() {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 16),
      child: Row(
        children: [
          FilterChip(
            label: const Text('All'),
            selected: _filter == 'all',
            onSelected: (_) {
              setState(() => _filter = 'all');
              _loadMerchants();
            },
          ),
          const SizedBox(width: 8),
          FilterChip(
            label: const Text('Pending'),
            selected: _filter == 'pending',
            onSelected: (_) {
              setState(() => _filter = 'pending');
              _loadMerchants();
            },
          ),
          const SizedBox(width: 8),
          FilterChip(
            label: const Text('Approved'),
            selected: _filter == 'approved',
            onSelected: (_) {
              setState(() => _filter = 'approved');
              _loadMerchants();
            },
          ),
        ],
      ),
    );
  }

  Widget _buildMerchantsList() {
    if (_loading) return const Center(child: CircularProgressIndicator());
    if (_error != null) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            const Icon(Icons.error_outline, size: 48, color: Colors.red),
            const SizedBox(height: 12),
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 32),
              child: Text(
                'Failed to load P2P merchants: $_error',
                textAlign: TextAlign.center,
                style: TextStyle(color: Colors.grey[600]),
              ),
            ),
            const SizedBox(height: 12),
            ElevatedButton(onPressed: _loadMerchants, child: const Text('Retry')),
          ],
        ),
      );
    }
    if (_merchants.isEmpty) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            const Icon(Icons.store_mall_directory_outlined, size: 48, color: Colors.grey),
            const SizedBox(height: 12),
            Text('No P2P merchants found', style: TextStyle(color: Colors.grey[600])),
          ],
        ),
      );
    }
    return ListView.builder(
      padding: const EdgeInsets.all(16),
      itemCount: _merchants.length,
      itemBuilder: (context, index) {
        final merchant = _merchants[index];
        return Card(
          margin: const EdgeInsets.only(bottom: 12),
          child: Padding(
            padding: const EdgeInsets.all(16),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Text('${merchant['business_name'] ?? '—'}', style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 16)),
                    Chip(
                      label: Text('${merchant['status'] ?? ''}'.toString().toUpperCase()),
                      backgroundColor: _getStatusColor(merchant['status']?.toString() ?? ''),
                    ),
                  ],
                ),
                const SizedBox(height: 8),
                Text('${merchant['email'] ?? '—'} • ${merchant['country'] ?? '—'}'),
                const SizedBox(height: 8),
                Row(
                  children: [
                    _buildDetail(
                      'Volume',
                      merchant['total_volume'] != null
                          ? '\$${((merchant['total_volume'] as num) / 1000).toStringAsFixed(1)}K'
                          : '—',
                    ),
                    _buildDetail('Txns', '${merchant['transaction_count'] ?? '—'}'),
                    _buildDetail('Rating', '${merchant['rating'] ?? '—'} ★'),
                  ],
                ),
                if (merchant['status'] == 'pending') ...[
                  const SizedBox(height: 12),
                  Row(
                    children: [
                      ElevatedButton(
                        onPressed: () => _approveMerchant(merchant['id'].toString()),
                        style: ElevatedButton.styleFrom(backgroundColor: Colors.green),
                        child: const Text('Approve'),
                      ),
                      const SizedBox(width: 8),
                      ElevatedButton(
                        onPressed: () => _rejectMerchant(merchant['id'].toString()),
                        style: ElevatedButton.styleFrom(backgroundColor: Colors.red),
                        child: const Text('Reject'),
                      ),
                    ],
                  ),
                ],
              ],
            ),
          ),
        );
      },
    );
  }

  Widget _buildDetail(String label, String value) {
    return Expanded(
      child: Column(
        children: [
          Text(label, style: TextStyle(fontSize: 11, color: Colors.grey[600])),
          Text(value, style: const TextStyle(fontWeight: FontWeight.w600)),
        ],
      ),
    );
  }

  Color _getStatusColor(String status) {
    switch (status) {
      case 'approved':
        return Colors.green;
      case 'pending':
        return Colors.orange;
      case 'rejected':
        return Colors.red;
      default:
        return Colors.grey;
    }
  }
}

File: admin/android/app/src/main/java/com/tigerwallet/admin/ui/screens/p2p_merchant/P2PMerchantScreen.kt
package com.tigerwallet.admin.ui.screens.p2p_merchant

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch

data class P2PMerchant(val id: Long, val businessName: String, val email: String, val country: String, val totalVolume: Double, val transactionCount: Int, val rating: Double, val status: String)

class P2PMerchantViewModel : ViewModel() {
    private val _merchants = mutableStateOf<List<P2PMerchant>>(emptyList())
    val merchants: List<P2PMerchant> = _merchants.value
    private val _loading = mutableStateOf(false)
    val loading: Boolean = _loading.value
    private val _filter = mutableStateOf("all")
    val filter: String = _filter.value
    private val _isDark = mutableStateOf(false)
    val isDark: Boolean = _isDark.value

    init { loadMerchants() }
    fun loadMerchants() { viewModelScope.launch { _loading.value = true; try { _merchants.value = emptyList() } catch (e: Exception) { } finally { _loading.value = false } }
    fun setFilter(f: String) { _filter.value = f; loadMerchants() }
    fun toggleTheme() { _isDark.value = !_isDark.value }
    fun approve(id: Long) { loadMerchants() }
    fun reject(id: Long) { loadMerchants() }

    // getMockMerchants removed: do not fabricate P2P merchants.
    )
}

@Composable
fun P2PMerchantScreen(viewModel: P2PMerchantViewModel = androidx.lifecycle.viewModel.compose.rememberViewModel { P2PMerchantViewModel() }) {
    val isDark = viewModel.isDark
    val merchants = viewModel.merchants
    val loading = viewModel.loading
    val filter = viewModel.filter
    val colors = if (isDark) darkColors() else lightColors()

    MaterialTheme(colors = colors) {
        Scaffold(topBar = { TopAppBar(title = { Text("P2P Merchants") }, actions = { IconButton(onClick = { viewModel.toggleTheme() }) { Icon(if (isDark) "light_mode" else "dark_mode", "Theme") } }, backgroundColor = colors.surface) }, backgroundColor = colors.background) { p ->
            Column(Modifier.padding(p)) {
                Row(Modifier.padding(16.dp).fillMaxWidth()) {
                    StatCard("Total", "${merchants.size}", Modifier.weight(1f))
                    StatCard("Approved", "${merchants.count { it.status == "approved" }}", Modifier.weight(1f))
                    StatCard("Pending", "${merchants.count { it.status == "pending" }}", Modifier.weight(1f))
                    StatCard("Volume", "$${(merchants.sumOf { it.totalVolume } / 1000).toInt()}K", Modifier.weight(1f))
                }
                Row(Modifier.padding(horizontal = 16.dp)) {
                    FilterChip(selected = filter == "all", onClick = { viewModel.setFilter("all") }, label = { Text("All") })
                    Spacer(Modifier.width(8.dp))
                    FilterChip(selected = filter == "pending", onClick = { viewModel.setFilter("pending") }, label = { Text("Pending") })
                    Spacer(Modifier.width(8.dp))
                    FilterChip(selected = filter == "approved", onClick = { viewModel.setFilter("approved") }, label = { Text("Approved") })
                }
                if (loading) Box(Modifier.fillMaxSize(), Alignment.Center) { CircularProgressIndicator() }
                else LazyColumn(Modifier.padding(16.dp)) {
                    items(merchants) { m ->
                        Card(Modifier.fillMaxWidth().padding(bottom = 12.dp)) {
                            Column(Modifier.padding(16.dp)) {
                                Row(Modifier.fillMaxWidth(), Arrangement.SpaceBetween) {
                                    Text(m.businessName, style = MaterialTheme.typography.titleMedium)
                                    Surface(color = when (m.status) { "approved" -> android.graphics.Color.GREEN; "pending" -> android.graphics.Color.parseColor("#FFA500"); else -> android.graphics.Color.GRAY }, shape = MaterialTheme.shapes.small) { Text(m.status.uppercase(), Modifier.padding(horizontal = 8.dp, vertical = 4.dp), color = android.graphics.Color.WHITE) }
                                }
                                Spacer(Modifier.height(8.dp))
                                Text("${m.email} • ${m.country}")
                                Row(Modifier.fillMaxWidth()) {
                                    Col("Volume", "$${(m.totalVolume / 1000).toInt()}K", Modifier.weight(1f))
                                    Col("Txns", "${m.transactionCount}", Modifier.weight(1f))
                                    Col("Rating", "${m.rating} ★", Modifier.weight(1f))
                                }
                                if (m.status == "pending") {
                                    Spacer(Modifier.height(8.dp))
                                    Row { Button(onClick = { viewModel.approve(m.id) }, colors = ButtonDefaults.buttonColors(backgroundColor = android.graphics.Color.GREEN)) { Text("Approve") }; Spacer(Modifier.width(8.dp)); Button(onClick = { viewModel.reject(m.id) }, colors = ButtonDefaults.buttonColors(backgroundColor = android.graphics.Color.RED)) { Text("Reject") } }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

@Composable private fun StatCard(label: String, value: String, m: Modifier) = Card(m = m.padding(4.dp)) { Column(Modifier.padding(12.dp), Alignment.CenterHorizontally) { Text(value, style = MaterialTheme.typography.headlineSmall); Text(label, style = MaterialTheme.typography.caption) } }
@Composable private fun Col(label: String, value: String, m: Modifier) = Column(m, Alignment.CenterHorizontally) { Text(label, style = MaterialTheme.typography.caption); Text(value) }
